### PR TITLE
Add mpd_send_save_queue, mpd_run_save_queue

### PR DESCRIPTION
**Pull request by jcorporation**
Adds support for the new optional arguments for the save command.

### CODE REVIEW

File: src/cplaylist.c [hunk 1]
```diff
@@ -178,6 +179,52 @@ mpd_run_save(struct mpd_connection *connection, const char *name)
...
+mpd_lookup_queue_save_mode(enum mpd_queue_save_mode mode)
+{
+	switch (mode) {
+	case MPD_QUEUE_SAVE_MODE_CREATE:
+		return "create";
+	case MPD_QUEUE_SAVE_MODE_REPLACE:
+		return "replace";
+	case MPD_QUEUE_SAVE_MODE_APPEND:
+		return "append";
+	default:
```
Comment (MaxKellermann) [inline src/cplaylist.c:204]: This `default:` disables the compiler warning when a new enum gets added, but we shouldn't suppress that warning

File: include/mpd/playlist.h [hunk 2]
```diff
@@ -59,6 +59,25 @@ struct mpd_connection;
...
+	MPD_QUEUE_SAVE_MODE_CREATE = 0,
+
+	/** Replace existing playlist */
+	MPD_QUEUE_SAVE_MODE_REPLACE,
+
+	/** Append to existing playlist */
+	MPD_QUEUE_SAVE_MODE_APPEND,
+
+	/** Unknown state */
+	MPD_QUEUE_SAVE_MODE_UNKNOWN
```
Comment (MaxKellermann) [inline include/mpd/playlist.h:78]: What does that mean?

File: src/cplaylist.c (hunk 1 above)
Comment (jcorporation) [reply src/cplaylist.c]: Correct, I have not taken this into account. Fixed in the commit.

File: include/mpd/playlist.h (hunk 2 above)
Comment (jcorporation) [reply include/mpd/playlist.h:78]: Incorrect description of this enum, fixed.

**Comment by jcorporation** [comment]
Thanks for review. I fixed the points and rebased the pull request to current master.

File: include/mpd/playlist.h [hunk 3]
```diff
@@ -374,6 +393,59 @@ mpd_send_save(struct mpd_connection *connection, const char *name);
...
+
+/**
+ * Parse the string to check which queue save mode it contains.
+ *
+ * @return the queue save mode enum
+ *
+ * @since libmpdclient 2.21.
+ */
+enum mpd_queue_save_mode
+mpd_parse_queue_save_mode(const char *mode);
```
[the remark this answers is not in the document]
Comment (jcorporation) [reply include/mpd/playlist.h:447]: I like the completeness of parse function if there is a lookup function. I looked into the source code of my MPD client implementation and I use it. I can remove the function here and move it into my code, if you think this is only bloat.

[the remark this answers is not in the document]
Comment (MaxKellermann) [reply include/mpd/playlist.h:447]: I believe it doesn't belong in libmpdclient, but if you really use it, I won't let you remove it now.

**Pull request merged**